Add script to generate DICOM thumbnails

fbudin69500 opened the pull request:
`process_dicom.py` is a Python script based on ITK that creates
slice thumbnails.
* The user specifies an input folder containing a DICOM series.
* The user specifies an output folder in which the generated files
will be saved.
* The number of slices can be specified (default is 10).
* Width and height of generated thumbnails can be specified (default is 512).
* If only one size (width or height) is given, the other parameter will
be computed such that the image aspect ratio is preserved.
* The thumbnails are saved as 8-bits JPEG images
* A JSON file describing the output file is generated.

Note: If multiple DICOM series are contained in the input folder, only
the first series will be used to generate the thumbnails.

Commit:
Add script to generate DICOM thumbnails

`process_dicom.py` is a Python script based on ITK that creates
slice thumbnails.
* The user specifies an input folder containing a DICOM series.
* The user specifies an output folder in which the generated files
will be saved.
* The number of slices can be specified (default is 10).
* Width and height of generated thumbnails can be specified (default is 512).
* If only one size (width or height) is given, the other parameter will
be computed such that the image aspect ratio is preserved.
* The thumbnails are saved as 8-bits JPEG images
* A JSON file describing the output file is generated.

Note: If multiple DICOM series are contained in the input folder, only
the first series will be used to generate the thumbnails.

fbudin69500 commented:
Note: The original `Dockerfile` was renamed `Dockerfile.volume` and a new `Dockerfile` named `Dockerfile.dicom` was created to create a docker image to run the new `process_dicom.py` script.

thewtex commented on preprocess/process_dicom.py:
```diff
@@ -0,0 +1,294 @@
+        # used to apply an intensity windowing filter.
+        width = int(meta_dict['0028|1051'])
+    except KeyError:
+        raise Exception("DICOM window `center` or `width` tag not found.")
```
In case the window / level is not in the DICOM tags (`center` and `width` will not be defined below), a `RescaleIntensityImageFilter` could be used. This will ensure a mapping to 0 to 255 from what may be a broader range.

fbudin69500 replied:
Yes, this is a limitation that I knew, and I was thinking that the way to improve this would be to rescale between min and max of the image. A rescaling between 0 and 255 is also performed at the end before the image is saved, although thinking about it now, this is performed on a slice by slice bases and is not optimal. I will improve this.

fbudin69500 replied:
The new commit should address your concerns.

Commit:
Rescaling is performed on 3D image instead of on each individual slices

Image intensity rescaling is performed directly on the 3D image. This allows
to rescale the image intensity using the image minimum and maximum if
the DICOM `window width` and `window_center` are not available. This also
avoid any risk of rescaling differently each slice.

thewtex commented on preprocess/process_dicom.py:
```diff
@@ -129,28 +129,31 @@ def rescale_dicom_image_intensity(image, meta_dict):
+        # we simply rely on the image minimum and maximum to rescale it.
+        # Image is rescaled between 0 and 255 to be saved as a JPEG later.
+        return itk.RescaleIntensityImageFilter(image, OutputMinimum=0,
+                                               OutputMaximum=255)
```
An output image explicitly specified to have a pixel type of `unsigned char` will ensure the correct output type and will automatically specify `OutputMinimum` and `OutputMaximum`.

fbudin69500 replied:
True, but I was on purpose not converting the pixel type to `unsigned char` at this point of the script, before resampling, to keep all the image values as long as possible (until the last step).

zachmullen reviewed (approved):
@fbudin69500 is this new image in docker hub already?

fbudin69500 commented:
I haven't pushed it on dockerhub
